src/Uno.UI/Microsoft/UI/Xaml/Controls/ColorPicker/ColorSpectrum.cs

```diff
@@ -0,0 +1,1807 @@
+			ArrayList<byte> bgraMaxPixelData = new ArrayList<byte>();
+			List<Hsv> newHsvValues = new List<Hsv>();
+
+			// Uno Docs: size_t not available in C# so types were changed
+			var pixelCount = (int)(Math.Round(minDimension) * Math.Round(minDimension));
+			var pixelDataSize = pixelCount * 4;
+			bgraMinPixelData.Capacity = pixelDataSize;
+
+			// We'll only save pixel data for the middle bitmaps if our third dimension is hue.
+			if (components == ColorSpectrumComponents.ValueSaturation ||
+				components == ColorSpectrumComponents.SaturationValue)
+			{
+				bgraMiddle1PixelData.Capacity = pixelDataSize;
+				bgraMiddle2PixelData.Capacity = pixelDataSize;
+				bgraMiddle3PixelData.Capacity = pixelDataSize;
+				bgraMiddle4PixelData.Capacity = pixelDataSize;
+			}
+
+			bgraMaxPixelData.Capacity = pixelDataSize;
+			newHsvValues.Capacity = pixelCount;
+
+			int minDimensionInt = (int)Math.Round(minDimension);
+			//WorkItemHandler workItemHandler = 
+			//(IAsyncAction workItem) =>
+			await Task.Run(() =>
+			{
+				// As the user perceives it, every time the third dimension not represented in the ColorSpectrum changes,
+				// the ColorSpectrum will visually change to accommodate that value.  For example, if the ColorSpectrum handles hue and luminosity,
+				// and the saturation externally goes from 1.0 to 0.5, then the ColorSpectrum will visually change to look more washed out
+				// to represent that third dimension's new value.
+				// Internally, however, we don't want to regenerate the ColorSpectrum bitmap every single time this happens, since that's very expensive.
+				// In order to make it so that we don't have to, we implement an optimization where, rather than having only one bitmap,
+				// we instead have multiple that we blend together using opacity to create the effect that we want.
+				// In the case where the third dimension is saturation or luminosity, we only need two: one bitmap at the minimum value
+				// of the third dimension, and one bitmap at the maximum.  Then we set the second's opacity at whatever the value of
+				// the third dimension is - e.g., a saturation of 0.5 implies an opacity of 50%.
+				// In the case where the third dimension is hue, we need six: one bitmap corresponding to red, yellow, green, cyan, blue, and purple.
+				// We'll then blend between whichever colors our hue exists between - e.g., an orange color would use red and yellow with an opacity of 50%.
+				// This optimization does incur slightly more startup time initially since we have to generate multiple bitmaps at once instead of only one,
+				// but the running time savings after that are *huge* when we can just set an opacity instead of generating a brand new bitmap.
+				if (shape == ColorSpectrumShape.Box)
+				{
+					for (int x = minDimensionInt - 1; x >= 0; --x)
+					{
+						for (int y = minDimensionInt - 1; y >= 0; --y)
+						{
+							//if (workItem.Status == AsyncStatus.Canceled)
```

Comment:
Codacy found an issue: [Remove this commented out code.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=6121259)